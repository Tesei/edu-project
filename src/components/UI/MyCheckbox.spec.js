import { describe, it, expect, beforeAll, beforeEach, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import component from './MyCheckbox.vue'

describe('MyCheckbox component', () => {
    const buildWrapper = (slotText = 'Main Content', error = false) => {
        const propsData = {
            error,
        }
        return mount(component, {
            slots: {
                default: slotText,
            },
            propsData,
        })
    }
    it('mounts without error', () => {
        const wrapper = buildWrapper()
        expect(wrapper.exists).toBeTruthy()
        // expect(wrapper.html()).toMatchSnapshot()
    })
    it('slot set html', () => {
        const wrapper = buildWrapper('<div>Main Content</div>')
        expect(wrapper.find('label').html()).toContain('<div>Main Content</div>')
    })
    it('slot set text', () => {
        const wrapper = buildWrapper('Main Content')
        expect(wrapper.find('label').text()).toContain('Main Content')
    })
    it('if checkbox has props error', () => {
        const wrapper = buildWrapper('', true)
        expect(wrapper.find('label').attributes().class).includes('_red')
    })

    it('should update isChecked when checkbox is checked', async () => {
        const wrapper = buildWrapper('', true)
        const checkbox = wrapper.find('input[type="checkbox"]')

        await checkbox.setChecked()

        expect(wrapper.vm.isChecked).toBe(true)
    })
    it('should update isChecked when checkbox is unchecked', async () => {
        const wrapper = buildWrapper()
        const checkbox = wrapper.find('input[type="checkbox"]')

        await checkbox.setChecked()
        await checkbox.setChecked(false)

        expect(wrapper.vm.isChecked).toBe(false)
    })
})
