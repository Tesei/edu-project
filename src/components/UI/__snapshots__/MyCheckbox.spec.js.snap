// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > mounts without error 1`] = `"<div data-v-bc73ac9a="" class="my-checkbox"><input data-v-bc73ac9a="" id="0.4249740080217943" type="checkbox" class="my-checkbox__input"><label data-v-bc73ac9a="" class="my-checkbox__label" for="0.4249740080217943">Main Content</label></div>"`;
